sashimi-webapp/src/database/data-modifier/dataAdd.js

```diff
@@ -1,26 +1,82 @@
-import exceptions from '../exceptions';
+import exceptions from 'src/database/exceptions';
+import constants from 'src/database/constants';
+import SqlCommands from 'src/database/sql-related/sqlCommands';
+import DateTime from 'src/database/generated-data/dateTime';
+import SqlArray from 'src/database/generated-data/sqlArray';
+import StringManipulator from 'src/database/stringManipulation';
 
-import constants from '../constants';
+const sqlCommands = new SqlCommands();
+const dateTime = new DateTime();
+const alasqlArray = new SqlArray();
+const stringManipulator = new StringManipulator();
 
-import SqlCommands from '../sql-related/sqlCommands';
+function isLowerUpperBoundWithinRange(lowerBound, upperBound, givenArray) {
+  return lowerBound < upperBound
+    && lowerBound >= 0
+    && upperBound < givenArray.length;
+}
 
-import DateTime from '../generated-data/dateTime';
+function exchange(givenArray, currIndex, thenIndex) {
```

Comment:
If this is meant to be a generic function for swapping elements in an array, maybe it might be better to rename the parameters `currIndex` and `thenIndex` to something like `index1` and `index2`?